Controllers/BranchesController.cs

```diff
@@ -0,0 +1,215 @@
+        public async Task<IActionResult> Edit(int? id)
+        {
+            if (id == null)
+            {
+                return NotFound();
+            }
+
+            var branch = await _context.Branches
+                .Include(b => b.Contact)
+                .Include(b => b.Entity)
+                .FirstOrDefaultAsync(m => m.Id == id);
+
+            //var branch = await _context.Branches.FindAsync(id);
+
+            if (branch == null)
+            {
+                return NotFound();
+            }
+            ViewData[nameof(Branch.EntityId)] = new SelectList(_context.Entities, nameof(Entity.Id), nameof(Entity.Name));
+
+            var branchVM = _mapper.Map<BranchFormViewModel>(branch);
+
+            return View(branchVM);
+        }
+
+        // POST: Branches/Edit/5
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public async Task<IActionResult> Edit(int id, BranchFormViewModel branchVM)
+        {
+            if (id != branchVM.Id)
+            {
+                return NotFound();
+            }
+
+            if (ModelState.IsValid)
+            {
+                try
+                {
+                    var branch = _mapper.Map<Branch>(branchVM);
+                    _context.Update(branch);
+                    await _context.SaveChangesAsync();
+                }
+                catch (DbUpdateConcurrencyException)
+                {
+                    if (!BranchExists(branchVM.Id))
+                    {
+                        return NotFound();
+                    }
+                    else
+                    {
+                        throw;
+                    }
+                }
+                return RedirectToAction(nameof(Index));
+            }
+            ViewData[nameof(Branch.EntityId)] = new SelectList(_context.Entities, nameof(Entity.Id), nameof(Entity.Name));
+
+            return View(branchVM);
+        }
+
+        // GET: Branches/Delete/5
+        public async Task<IActionResult> Delete(int? id)
+        {
+            if (id == null)
+            {
+                return NotFound();
+            }
+
+            var branch = await _context.Branches
+                .Include(b => b.Contact)
+                .Include(b => b.Entity)
+                .FirstOrDefaultAsync(m => m.Id == id);
+            if (branch == null)
+            {
+                return NotFound();
+            }
+
+            var branchVM = _mapper.Map<BranchFormViewModel>(branch);
+
+            return View(branchVM);
+        }
+
+        // POST: Branches/Delete/5
+        [HttpPost, ActionName("Delete")]
+        [ValidateAntiForgeryToken]
+        public async Task<IActionResult> DeleteConfirmed(int id)
+        {
+            var branch = await _context.Branches.FindAsync(id);
+            _context.Branches.Remove(branch);
+            await _context.SaveChangesAsync();
+            return RedirectToAction(nameof(Index));
+        }
+
+        private bool BranchExists(int id)
```

Comment:
You can use AnyAsync as well here as Any will perform IO, so there is point to do this asynchronous.

Reply:
When you call BranchExists(...) you do roundtrip to database performing sql check against db. This operation is considered as IO (input - output). As IO block thread until response comes back it is worth to make this async. Async opertion cause you release _thread_ to _therad pool_ as IO operation spent some longer time waiting for response from db. In this time there are more free threads for your server apllication and your app can serve content to other users by handling other request. As IO finish another thread will be employ to finish work.

So what I ask you here is to rewrite interface of this method:
`private bool BranchExists(int id)`
to
`private async Task<bool> BranchExists(int id)`
and use `AnyAsync` instead of `Any` extension method over `DbContext`.